### DISCUSSION TITLE
Suggestion : Compare type between related tables

### DISCUSSION THREAD

**Discussion by mortenseifert** [StefanMaron/BusinessCentral.LinterCop#287] [New Rule]
When approving pull requests, it is not easy to see if there is a conflict between data type on related tables.

E.g. on Sales Header I add the following fields
field(50.000; "Field A"; Integer)
field(50.002; "Field B"; Code[10])

In the code, I already have Sales Shipment Header
field(50.000; "Field A"; Integer)
field(50.001; "Field B"; Code[10])
Field(50.002; "My Date"; Date)

If the LinterCop could warn about the type in field 50.002 for Sales Header / Sales Shipment Header was not the same, it would be really great.

From the top of my head the tables to be examined should be
Sales Header = Sales Shipment Header = Sales Invoice Header = Sales Cr. Memo Header = Return Receipt Header
Purchase Header = Purch. Rcpt. Header = Purch. Inv. Header = Purch. Cr. Memo Hdr. = Return Receipt Header
Contact = Customer = Vendor

And many more
* Service
* Finance Charge
* etc.

**Comment by mortenseifert** [comment]
If I can assist in any way, I would love to contribute.

**Reply by Arthurvdv** [reply]
I have good news, @tinestaric has created the [LC0044](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0044) `TransferFields consistency between records.` rule, which is now available in the (pre)release [v0.30.10](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.10) version of the LinterCop.

**Reply by mortenseifert** [reply]
This is fantastic. Just tries the pre-view and it works really nice 🙏
